docs/pnpm.md

```diff
@@ -258,7 +260,15 @@ npm packages have "lifecycle scripts" such as `postinstall` which are documented
 
 We refer to these as "lifecycle hooks".
 
-> You can disable this feature completely by setting all packages to have no hooks, using
+Determining which packages have lifecycle hooks is determined by, in priority order:
+
+1. The workspace `package.json` [`pnpm.onlyBuiltDependencies` attribute](https://pnpm.io/package_json#pnpmonlybuiltdependencies).
+2. The pnpm-lock.yaml `requiresBuild` attribute. Note this is only available in pnpm _before_ v9, see [pnpm #7707](https://github.com/pnpm/pnpm/issues/7707) for reasons why this attribute was removed.
```

Comment:
Reading the code it looks like if `pnpm.onlyBuiltDependencies` is set then `requiresBuild` is ignored:

```
run_lifecycle_hooks = name in only_built_dependencies if only_built_dependencies != None else requires_build
```

That sounds right since if the user wants to use `requilresBuild` from pre-pnpm v9 then they don't set any `onlyBuiltDependencies`. I think this could be made clearer here in the docs.

What happens in the corner case of the user setting `onlyBuiltDependencies` to an empty list `[]` in the package.json?

Reply:
Also, if the user sets `onlyBuiltDependencies` in pnpm v9, does the `requiresBuild` in the lock file reflect the user's choices or is that unchanged?

Reply:
> Reading the code it looks like if pnpm.onlyBuiltDependencies is set then requiresBuild is ignored:

Correct. Using `onlyBuiltDependencies` sets an explicit list of packages that have lifecycle hooks, falling back to the `requiresBuild` flag if the `onlyBuiltDependencies` is not set.

Where `npm_translate_lock(lifecycle_hooks)` is a *filter* of those lifecycle hooks.

> What happens in the corner case of the user setting onlyBuiltDependencies to an empty list [] in the package.json?

Leaving `onlyBuiltDependencies` unset implies using the "default" behaviour, which for pnpm v9+ means no lifecycle hooks at all 😢

Reply:
Is there a difference for pnpm v8 if you omit `onlyBuiltDependencies` entirely and if you set it to an empty list `[]`. From reading the code I think in the former you'd get a `None` and in the latter you'd get an empty `[]` list out of the `package.json` read which would be different behaviors?

Reply:
In the `e2e/pnpm_lockfiles` if I add `onlyBuiltDependencies: []` all 3 lockfile versions seem to do the same thing: all lifecycle hook targets are gone.

Reply:
I'm thinking once we had v9 we. can edit that test to specify `onlyBuiltDependencies: ["@aspect-test/c"]`. Then it might be nice to assert that things like `.aspect_rules_js/node_modules/fsevents@2.3.3/lc` do *not* have lifecycle hooks, but asserting targets do not exist is a bit more of a hassle 🤷

Reply:
Cool. That answer my question 👍